CenterForOpenScience/osf.io
# [Feature] Replace wbv0 `...url_for` with wbv1 [OSF-8696]

## Purpose
WBV0 is kill.


## Changes
Replace calls to `waterbutler_url_for` with calls to `waterbutler_api_url_for`


## Side effects
Removed skip logic from test that said "Unskip when figshare addon is implemented" from... 10 months ago. 



## Ticket

https://openscience.atlassian.net/browse/OSF-8696

## QA notes
1. Make sure you can still upload conference files via email. I only could half test this since I'm not sure how to receive emails locally.
2. Make sure that files and folders (and addons) are still archived to osfstorage properly when registered
